Celigo.ServiceManager.NetSuite.REST/src/RESTletClient.cs

```diff
@@ -20,11 +21,62 @@ public struct Passport
     
     public interface IRestletClient
     {
-        Task<HttpResponseMessage> Get(in string account, in string token, in string tokenSecret, params (string key, string value)[] queryParams);
-        Task<HttpResponseMessage> Post<T>(in string account, in string token, in string tokenSecret, in T message, params (string key, string value)[] queryParams);
+        Task<HttpResponseMessage> Get(
+            in string account, 
+            in string token, 
+            in string tokenSecret, 
+            IReadOnlyDictionary<string, string> queryParams = null
+        );
         
-        Task<HttpResponseMessage> Get(in Passport passport, params (string key, string value)[] queryParams);
-        Task<HttpResponseMessage> Post<T>(in Passport passport, in T message, params (string key, string value)[] queryParams);
+        Task<HttpResponseMessage> Post<T>(
+            in string account, 
+            in string token, 
+            in string tokenSecret, 
+            in T message, 
+            IReadOnlyDictionary<string, string> queryParams = null
+        );
+        
+        Task<HttpResponseMessage> Get(
+            in Passport passport, 
+            IReadOnlyDictionary<string, string> queryParams = null
+        );
+        
+        Task<HttpResponseMessage> Post<T>(
+            in Passport passport, 
+            in T message, 
+            IReadOnlyDictionary<string, string> queryParams = null
+        );
+        
+        [Obsolete]
+        Task<HttpResponseMessage> Get(
+            in string account, 
+            in string token, 
+            in string tokenSecret, 
+            (string key, string value) queryParam,
+            params (string key, string value)[] queryParams
```

Comment:
Good question :) @somenathmaji 

In a simplified form, we used to have the following method signatures:
```
Mthd(params SomeType[] queryParams)
```

We wanted to deprecate these and add
```
Mthd(AnotherType queryParams = null)
```

Both of these signatures mean that the caller can invoke this method without parameters
```
Mthd() // valid because params array can be empty and in the other signature, parameter is explicitly optional.
```

So the above would be an ambiguous call for .NET.
Given that anyone had previously used the tuple based signature, they would have at least written
```
Mthd((k, v));
```
the approach is able to
 1. Disambiguate the two signatures
 2. Point the complier to the latest version of the method (which is `Mthd(AnotherType...)`
 3. Not break any code that was already using tuple based approach.

Reply:
Right!! understood.